packages/components/src/components/hds/rich-tooltip/toggle.ts

```diff
@@ -6,11 +6,42 @@
 import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
 
-export const DEFAULT_ICONPOSITION = 'trailing';
-export const ICONPOSITIONS = ['leading', 'trailing'];
-export const SIZES = ['small', 'medium', 'large'];
+import {
+  HdsRichTooltipToggleIconPositionValues,
+  HdsRichTooltipToggleSizeValues,
+} from './types.ts';
+import type {
+  HdsRichTooltipToggleIconPositions,
+  HdsRichTooltipToggleSizes,
+} from './types.ts';
+import type { FlightIconSignature } from '@hashicorp/ember-flight-icons/components/flight-icon';
+import type { HdsPopoverPrimitiveSignature } from '../popover-primitive';
 
-export default class HdsRichTooltipToggleComponent extends Component {
+export const ICONPOSITIONS: string[] = Object.values(
+  HdsRichTooltipToggleIconPositionValues
+);
+export const DEFAULT_ICONPOSITION =
+  HdsRichTooltipToggleIconPositionValues.Trailing;
+export const SIZES: string[] = Object.values(HdsRichTooltipToggleSizeValues);
+
+export interface HdsRichTooltipToggleSignature {
+  Args: {
+    text?: string;
+    icon?: FlightIconSignature['Args']['name'];
```

Comment:
I hadn't considered coupling the types like this, but I think it makes sense.

Reply:
Yes, for me that's the power of TS in a design system, the fact that you don't have to worry about changing types across components, when they are connected in this way you can be sure that a change gets automatically "propagated" to the consumers without having to remember to do it manually

Reply:
Makes sense!